lib/ox/element.rb

```diff
@@ -150,7 +150,18 @@ def locate(path)
     # _raise_ [NoMethodError] if no match is found
     def method_missing(id, *args, &block)
       has_some = false
-      ids = id.to_s
+      ids = id.to_s.sub('_', ':') # allows easy API with namespaces
```

Comment:
I understand now. Basically, XML allows more characters in element names than Ruby does for method names. How about use _ as you suggest but instead of a sub a try without the subject first and the try with the sub although since namespaces can be nested, gsub is better.